### PR TITLE
Add Audit Log documentation for Prefect Cloud

### PR DESCRIPTION
**Pull request by tpdorsey**
Adds documentation for [Audit Log](https://deploy-preview-7404--prefect-orion.netlify.app/ui/audit-log/).

To do:

- [ ] Audit Log events
- [ ] Screenshot of audit logs
- [ ] Update Cloud screenshots to include Audit Log

### Checklist

- [ ] This pull request references any related issue by including "closes `<link to issue>`"
	- If no issue exists and your change is not a small fix, please [create an issue](https://github.com/PrefectHQ/prefect/issues/new/choose) first.
- [x] This pull request includes tests or only affects documentation.
- [ ] This pull request includes a label categorizing the change e.g. `fix`, `feature`, `enhancement`

### CODE REVIEW

File: docs/ui/audit-log.md [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    - teams
+    - workspaces
+    - organizations
+    - audit logs
+    - compliance
+---
+
+# Audit Log <span class="badge cloud"></span> <span class="badge orgs"></span> <span class="badge enterprise"></span>
+
+Prefect Cloud's [Organization and Enterprise plans](https://www.prefect.io/pricing) offer compliance and transparency functionality with Audit Log in Prefect Cloud 2. Audit logs provide a chronological record of activities performed by Prefect Cloud users in your organization, allowing you to monitor detailed Prefect Cloud actions for security and compliance purposes.
```
Comment (billpalombi) [inline docs/ui/audit-log.md:17]:
```suggestion
Prefect Cloud's [Organization and Enterprise plans](https://www.prefect.io/pricing) offer enhanced compliance and transparency with Audit Log. Audit logs provide a chronological record of activities performed by Prefect Cloud users in your organization, allowing you to monitor detailed Prefect Cloud actions for security and compliance purposes. 
```

**Review by billpalombi — APPROVED** [review]
One minor tweak, otherwise lgtm

File: docs/ui/audit-log.md [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+Prefect Cloud's [Organization and Enterprise plans](https://www.prefect.io/pricing) offer enhanced compliance and transparency with Audit Log. Audit logs provide a chronological record of activities performed by Prefect Cloud users in your organization, allowing you to monitor detailed Prefect Cloud actions for security and compliance purposes. 
+
+Audit logs enable you to identify who took what action, when, and using what resources within your Prefect Cloud organization. In conjunction with appropriate tools and procedures, audit logs can assist in detecting potential security violations and investigating application errors.  
+
+Audit logs can be used to identify changes in: 
+
+- Access to workspaces
+- User login activity
+- API key creation 
+
```
Comment (nathanielaldon) [inline docs/ui/audit-log.md:26]:
```suggestion
- Workspace creation and removal
- Organization member invitations and removal
- Service account creation, API key rotation, and removal
- Billing payment method for self-serve pricing tiers

```

File: docs/ui/audit-log.md [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+![Viewing audit logs for an organization in the Prefect Cloud UI.](/img/ui/audit-log.png)
+
+Organization admins can view audit logs for: 
+
+- Organization-level events in Prefect Cloud (for example, inviting a user, changing a user’s organization role, or deleting a service account).
+- Workspace-level events in Prefect Cloud (for example, adding a user to a workspace, changing a user’s workspace role, or deleting a workspace).
+
+Admins can filter audit logs on multiple dimensions to restrict the results they see by workspace, user, or event type.
+
+Audit logs may also be filtered by date range. Audit log retention policy varies by [Prefect Cloud plan](https://www.prefect.io/pricing). See your [organization profile page](/ui/organizations/) for the current audit log retention period.
```
Comment (nathanielaldon) [inline docs/ui/audit-log.md:42]:
```suggestion
Audit logs may also be filtered by date range. Audit log retention period varies by [Prefect Cloud plan](https://www.prefect.io/pricing). See your [organization profile page](/ui/organizations/) for the current audit log retention period.
```

File: docs/ui/cloud.md [hunk 4]
```diff
@@ -98,6 +99,10 @@ See the [role-based access controls (RBAC)](/ui/roles/) documentation for more i
 
 Prefect Cloud's [Organization and Enterprise plans](https://www.prefect.io/pricing) offer [single sign-on (SSO)](/ui/sso/) authentication integration with your team’s identity provider. SSO integration can bet set up with identity providers that support OIDC and SAML.
 
+## Audit Log <span class="badge orgs"></span> <span class="badge enterprise"></span>
+
+Prefect Cloud's [Organization and Enterprise plans](https://www.prefect.io/pricing) offer [Audit Log](/ui/audit-log/) compliance and transparency functionality. Audit logs provide a chronological record of activities performed by users in your organization, allowing you to monitor detailed actions for security and compliance purposes.
```
Comment (nathanielaldon) [inline docs/ui/cloud.md:104]:
```suggestion
Prefect Cloud's [Organization and Enterprise plans](https://www.prefect.io/pricing) offer [Audit Log](/ui/audit-log/) compliance and transparency tools. Audit logs provide a chronological record of activities performed by users in your organization, allowing you to monitor detailed actions for security and compliance purposes. 
```

File: docs/ui/cloud.md [hunk 5]
```diff
@@ -33,6 +33,7 @@ Prefect Cloud is a workflow coordination-as-a-service platform. Prefect Cloud pr
     - [Service accounts](/ui/service-accounts/) &mdash; configure API access for running agents or executing flow runs on remote infrastructure.
     - [Custom role-based access controls (RBAC)](/ui/roles/) &mdash; assign users granular permissions to perform certain activities within an organization or a workspace.
     - [Single Sign-on (SSO)](/ui/sso/) authentication using your identity provider.
+    - [Audit Log](/ui/audit-log/) record of user activities for for security and compliance.
```
Comment (nathanielaldon) [inline docs/ui/cloud.md:36]:
```suggestion
    - [Audit Log](/ui/audit-log/) record of user activities to monitor security and compliance.
```

**Review by nathanielaldon — APPROVED** [review]
Added a few proposed wording tweaks, and fleshed out the use case list. Thanks a ton for putting all this together, Terry!

**Pull request merged**